src/components/Inputs/ClassKey.js

```diff
@@ -0,0 +1,42 @@
+import { FormSpy } from 'react-final-form'
+import React, { Fragment } from 'react'
+import i18n from '@dhis2/d2-i18n'
+import propTypes from 'prop-types'
+
+import { FORMAT_KEY, OPTION_CSV } from './Format'
+import { RadioGroup } from '../FinalFormComponents/RadioGroup'
+import { Select } from '../FinalFormComponents/Select'
+
+export const CLASS_KEY_KEY = 'classKey'
+
+export const ClassKey = ({ show, options, defaultValue }) => (
+    <Fragment>
+        {show && (
+            <Select
+                name={CLASS_KEY_KEY}
+                label={i18n.t('Class key')}
+                options={options}
+                defaultValue={defaultValue}
+            />
+        )}
+
+        <FormSpy
+            render={({ values, form }) => {
```

Comment:
They've hidden this well in the docs. `form.change` seems so much nicer than those hideous mutator functions

Reply:
Yes!